cmd/kubeadm/app/preflight/checks.go

```diff
@@ -967,11 +967,13 @@ func addCommonChecks(execer utilsexec.Interface, cfg kubeadmapi.CommonConfigurat
 
 	// non-windows checks
 	if runtime.GOOS == "linux" {
+		if !isDocker {
```

Comment:
Eventually I'd like to use crictl for talking to docker as well, hmm, I think requiring this always could be ok

Reply:
different versions of docker might not be able to respond properly to crictl. IMHO, it is safer to use crictl with explicitly cri-compatible runtime

Reply:
@kad could we put a //TODO comment there to re-evaluate based on docker version and compatibility, 
then lgtm imo 

/approve

Reply:
> Eventually I'd like to use crictl for talking to docker as well

You can do it if you specify full socket url pointing to the docker socket, i.e. --cri-socket unix:///var/run/dockershim.sock

Reply:
@luxas 
> hmm, I think requiring this always could be ok

Requesting crictl for the docker doesn't make sense as it's simply not used. And it's not harmless as it creates great confusion especially among docker users that even don't know what CRI is, but obliged to install crictl. Here are some examples:
 * https://github.com/kubernetes/kubeadm/issues/992
 * https://github.com/kubernetes/kubeadm/issues/657
 * https://github.com/kubernetes/kubeadm/issues/814
 * https://github.com/kubernetes/kubeadm/issues/700
 * https://github.com/kubernetes/kubeadm/issues/657

Reply:
Eventually, yes, it would be great to replace all container runtime calls with crictl.